src/test/java/api/loans/OverrideRenewByBarcodeTests.java

```diff
@@ -0,0 +1,824 @@
+      hasMessage("Cannot renew item checked out to different user"),
+      hasUserRelatedParameter(james))));
+  }
+
+  @Test
+  public void canOverrideRenewalToSpecifiedDateWhenDateFallsOutsideOfTheDateRangesInTheLoanPolicy() throws
+    InterruptedException,
+    ExecutionException,
+    TimeoutException, MalformedURLException {
+
+    final DateTime renewalDate = DateTime.now();
+
+    IndividualResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    final IndividualResource jessica = usersFixture.jessica();
+
+    final IndividualResource loan = loansFixture.checkOutByBarcode(smallAngryPlanet, jessica,
+      new DateTime(2018, 4, 21, 11, 21, 43));
+
+    final UUID loanId = loan.getId();
+
+    FixedDueDateSchedulesBuilder fixedDueDateSchedules = new FixedDueDateSchedulesBuilder()
+      .withName("Kludgy Fixed Due Date Schedule")
+      .addSchedule(wholeMonth(2018, 2))
+      .addSchedule(forDay(renewalDate));
+
+    final UUID fixedDueDateSchedulesId = fixedDueDateScheduleClient.create(
+      fixedDueDateSchedules).getId();
+
+    //Need to remember in order to delete after test
+    schedulesToDelete.add(fixedDueDateSchedulesId);
+
+    LoanPolicyBuilder currentDueDateRollingPolicy = new LoanPolicyBuilder()
+      .withName("Current Due Date Rolling Policy")
+      .rolling(Period.months(2))
+      .renewWith(fixedDueDateSchedulesId)
+      .renewFromCurrentDueDate();
+
+    UUID dueDateLimitedPolicyId = loanPolicyClient.create(currentDueDateRollingPolicy).getId();
+
+    //Need to remember in order to delete after test
+    policiesToDelete.add(dueDateLimitedPolicyId);
+
+    useLoanPolicyAsFallback(dueDateLimitedPolicyId);
+
+    DateTime newDueDate =
+      new DateTime(2025, 1, 1, 1, 1, 1);
+    final JsonObject renewedLoan =
+      loansFixture.overrideRenewalByBarcode(smallAngryPlanet, jessica, OVERRIDE_COMMENT, newDueDate.toString()).getJson();
+
+    assertThat(renewedLoan.getString("id"), is(loanId.toString()));
+
+    assertThat("user ID should match barcode",
+      renewedLoan.getString("userId"), is(jessica.getId().toString()));
+
+    assertThat("item ID should match barcode",
+      renewedLoan.getString("itemId"), is(smallAngryPlanet.getId().toString()));
+
+    assertThat("status should be open",
+      renewedLoan.getJsonObject("status").getString("name"), is("Open"));
+
+    assertThat("action should be renewed",
+      renewedLoan.getString("action"), is("Renewed through override"));
+
+    assertThat("'actionComment' field should contain comment specified for override",
+      renewedLoan.getString("actionComment"), is(OVERRIDE_COMMENT));
+
+    assertThat("renewal count should be incremented",
+      renewedLoan.getInteger("renewalCount"), is(1));
+
+    assertThat("last loan policy should be stored",
+      renewedLoan.getString("loanPolicyId"), is(dueDateLimitedPolicyId.toString()));
+
+    assertThat("due date should be 2 months after initial due date date",
```

Comment:
The overridden due date being checked is hard coded, however the description in the assertion does not reflect that, please change it to reflect the expectation (of 1st January 2025).

Reply:
description has been changed